src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int defaultCapacity = 16;
+    private final float loadFolder = 0.75f;
+    private int size = 0;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[defaultCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size != 0) {
+            checkCapacity();
+            table = putValue(key, value, table);
+        } else {
+            Node<K, V> newNode = new Node(hash(key), key, value, null);
+            table[index(table.length, newNode.hash)] = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[index(table.length, hash(key))];
+        if (null == currentNode) {
+            return null;
+        }
+        if ((null == key && null == currentNode.key)
+                || null != key && key.equals(currentNode.key)) {
+            return currentNode.value;
+        } else {
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+                if ((null == key && null == currentNode.key)
+                        || null != key && key.equals(currentNode.key)) {
+                    return currentNode.value;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkCapacity() {
+        if (size >= table.length * loadFolder) {
```

Comment:
```suggestion
        if (size >= table.length * loadFolder) {
```

move this check  to the put method, and rename this method for resize